app/qml/PositionMarker.qml

```diff
@@ -11,6 +11,42 @@ Item {
     property QgsQuick.PositionKit positionKit
     property color baseColor: InputStyle.highlightColor
     property bool withAccuracy: true
+    // animated position
+    property real animatedPosX
+    property real animatedPosY
+
+    function changePos() {
+        var posNext = positionKit.mapSettings.coordinateToScreen(positionKit.projectedPosition)
+
+        animateX.from = animatedPosX
+        animateX.to = posNext.x
+        animateX.restart()
+
+        animateY.from = animatedPosY
+        animateY.to = posNext.y
+        animateY.restart()
+    }
+
+    Component.onCompleted: {
+        animatedPosX =  positionKit.projectedPosition.x
+        animatedPosY = positionKit.projectedPosition.y
+    }
+
+    NumberAnimation {
+        id: animateX
+        target: positionMarker
+        properties: "animatedPosX"
+        duration: 1000
+        easing {type: Easing.Linear}
```

Comment:
I would suggest to use a shorter duration e.g. 500 so that the marker isn't constantly moving and `InOutQuad` easing curve for more pleasant animation